### ISSUE TITLE
Research a possible integration of `zecwallet-light-cli` into Zebra

### ISSUE THREAD

**Issue by oxarbitrage** [ZcashFoundation/zebra#4181]
[zecwallet-light-cli](https://github.com/adityapk00/zecwallet-light-cli) is a zcash command line wallet written in rust.

It was briefly tested against a lightwalletd server backed up by a zebra node in the context of https://github.com/ZcashFoundation/zebra/issues/3655#issuecomment-1090904596

However, `zecwallet-light-cli` is also a library which could be potentially used by zebra to run integration tests.

This ticket will be used to investigate if an integration will worth and how much will be needed.

**Comment by oxarbitrage** [comment]
`zecwallet-light-cli` is a library so it can theoretically be inserted as a dependency into zebra. However, doing this is a dependency nightmare. I am not going to describe the details but just mention that it uses old and modified version of librustzcash. Other dependencies are also so old that had major refactoring. I got lost in that task so i changed my approach to see we can still get something from this.

So i went the other way around and left the dependencies as they are for now and added a dirty test in the cli to connect to a lightwalletd server running locally backed with a zebra node. The code for this is at https://github.com/oxarbitrage/zecwallet-light-cli/pull/1

Now, i think those tests found some bugs, or they might be bugs in the tests themselves but i will like to make sure.

The test for `get_taddress_txids` fails with:

```
thread 'lightclient::zebra_tests::zebra_backed_lightwalletd' panicked at 'called `Result::unwrap()` on an `Err` value: Status { code: Unknown, message: "-32602: Invalid params: invalid type: map, expected a sequence.", metadata: MetadataMap { headers: {"server": "nginx/1.14.0 (Ubuntu)", "date": "Sat, 23 Apr 2022 20:35:52 GMT", "content-type": "application/grpc", "content-length": "0"} } }', lib/src/lightclient/zebra_tests.rs:63:71
```

And the test for `get_taddress_balance` fails with:

```
thread 'lightclient::zebra_tests::zebra_backed_lightwalletd' panicked at 'called `Result::unwrap()` on an `Err` value: Status { code: Unknown, message: "json: cannot unmarshal array into Go struct field ZcashdRpcReplyGetaddressbalance.Balance of type int64", metadata: MetadataMap { headers: {"server": "nginx/1.14.0 (Ubuntu)", "date": "Sat, 23 Apr 2022 20:40:09 GMT", "content-type": "application/grpc", "content-length": "0"} } }', lib/src/lightclient/zebra_tests.rs:68:63
```

This is against the current zebra `main`.

Didn't tested utxos call as it is not merged into main so is not available. When called, the error will be the following which is ok:

```
thread 'lightclient::zebra_tests::zebra_backed_lightwalletd' panicked at 'called `Result::unwrap()` on an `Err` value: Status { code: Unknown, message: "-32601: Method not found", metadata: MetadataMap { headers: {"server": "nginx/1.14.0 (Ubuntu)", "date": "Sat, 23 Apr 2022 20:50:37 GMT", "content-type": "application/grpc", "content-length": "0"} } }', lib/src/lightclient/zebra_tests.rs:76:58
```

there are also other calls that i didn't tested.

**Comment by oxarbitrage** [comment]
I tested a branch with `getaddressutxos` available. 

This not fail but it returns an empty response for a taddress that should have utxos available:

Code used:
```rust
    // build an utxo request
    let utxos =  GetAddressUtxosArg {
        addresses: vec!["t3Vz22vK5z2LcKEdg16Yv4FFneEL1zg9ojd".to_string()],
        start_height: 1,
        max_entries: 0,
    };

    // success but empty
    let response = client.get_address_utxos(utxos).await.unwrap().into_inner();
    println!("{:?}", response);
```

Output:

```
GetAddressUtxosReplyList { address_utxos: [] }
```

**Comment by teor2345** [comment]
#4130 could be a good way to find and fix these RPC response format bugs?

**Comment by teor2345** [comment]
> The test for `get_taddress_txids` fails with ... `Invalid params: invalid type: map, expected a sequence`

This is now bug #4216 

> And the test for `get_taddress_balance` fails with ... `json: cannot unmarshal array into Go struct field ZcashdRpcReplyGetaddressbalance.Balance of type int64`

This is fixed by PR #4217 

> This not fail but it returns an empty response for a taddress that should have utxos available:

This is now bug #4220.

If you can find a RPC query that reproduces the missing UTXOs above, please add it to #4220.

**Comment by oxarbitrage** [comment]
This ticket helped to find the above bugs but there is a way that we can do the same (https://github.com/ZcashFoundation/zebra/pull/4253) without using `zecwallet-light-cli`. We will not integrate in the short run.
